rxandroidble/src/main/java/com/polidea/rxandroidble/internal/util/IllegalOperationChecker.java

```diff
@@ -0,0 +1,135 @@
+                    BluetoothGattCharacteristic.PROPERTY_WRITE_NO_RESPONSE,
+                    BluetoothGattCharacteristic.PROPERTY_SIGNED_WRITE,
+                    BluetoothGattCharacteristic.PROPERTY_INDICATE,
+                    BluetoothGattCharacteristic.PROPERTY_NOTIFY})
+    @interface BluetoothGattCharacteristicProperty { }
+
+    protected IllegalOperationChecker(@BluetoothGattCharacteristicProperty int propertyBroadcast,
+                                      @BluetoothGattCharacteristicProperty int propertyRead,
+                                      @BluetoothGattCharacteristicProperty int propertyWriteNoResponse,
+                                      @BluetoothGattCharacteristicProperty int propertyWrite,
+                                      @BluetoothGattCharacteristicProperty int propertyNotify,
+                                      @BluetoothGattCharacteristicProperty int propertyIndicate,
+                                      @BluetoothGattCharacteristicProperty int propertySignedWrite) {
+        this.propertyBroadcast = propertyBroadcast;
+        this.propertyRead = propertyRead;
+        this.propertyWriteNoResponse = propertyWriteNoResponse;
+        this.propertyWrite = propertyWrite;
+        this.propertyNotify = propertyNotify;
+        this.propertyIndicate = propertyIndicate;
+        this.propertySignedWrite = propertySignedWrite;
+    }
+
+
+    /**
+     * This method checks whether the supplied characteristic possesses properties supporting the requested kind of operation, specified by
+     * the supplied bitmask.
+     * @param characteristic a {@link BluetoothGattCharacteristic} the operation is done on
+     * @param neededProperties properties required for the operation to be successfully completed
+     */
+    public void checkAnyPropertyMatches(BluetoothGattCharacteristic characteristic,
+                                        @BluetoothGattCharacteristicProperty int neededProperties) {
+        final int characteristicProperties = characteristic.getProperties();
+
+        if ((characteristicProperties & neededProperties) == 0) {
+            int[] possibleProperties = getPossibleProperties();
+            String message = String.format(
+                    Locale.getDefault(),
+                    "Characteristic %s supports properties:[%s] (%d) does not have any property matching [%s] (%d)",
+                    characteristic.getUuid(),
+                    propertiesIntToString(characteristicProperties, possibleProperties),
+                    characteristicProperties,
+                    propertiesIntToString(neededProperties, possibleProperties),
+                    neededProperties
+            );
+            handleMessage(message);
+        }
+    }
+
+    /**
+     * Method for handling the potential non-match message
+     * @param message message describing the
+     */
+    protected abstract void handleMessage(String message);
```

Comment:
I'd prefer the checker to return the model with the report, rather than to pass it to a method that needs to be overridden. @dariuszseweryn what do you think? It's easier to test then.